modules/gnome/overlay.nix

```
@@ -0,0 +1,40 @@
+{
+  lib,
+  pkgs,
+  config,
+  options,
+  ...
+}:
+let
+  theme = pkgs.callPackage ./theme.nix {
+    inherit (config.lib.stylix) colors;
+    inherit (config.stylix) inputs;
+  };
+in
+{
+  overlay =
+    lib.mkIf
+      (
+        config.stylix.enable
+        && options.stylix.targets ? gnome
+        && config.stylix.targets.gnome.enable
+      )
+      (
+        _: super: {
```

```suggestion
        pkgs: super:
        let
          theme = pkgs.callPackage ./theme.nix {
            inherit (config.lib.stylix) colors;
            inherit (config.stylix) inputs;
          };
        in
        {
```

Using `callPackages` from within the overlay should help avoid inf-recursion.